### ISSUE TITLE
Hovering over reacted emoji displays multiple contact methods in tooltip

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#18921]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___
Precondition: account should have multiple contact methods

## Action Performed:

1. Sign in with account A
2. Add account B as a secondary login in the contact methods section
3. Send a message to account C
4. Change default contact method to secondary login account B
5. React to the sent message with emoji
6. Switch to the tab (window) where account C is signed in
7. Open the newly sent message
8. Refresh the page for the change to take effect
9. Hover over the emoji reaction

## Expected Result: 
The tooltip should display one email address (default contact method)
## Actual Result: 
The tooltip displays multiple contact methods
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.13.3
**Reproducible in staging?:** y
**Reproducible in production?:**
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/43996225/68dfbdf3-a1fc-407c-8ffe-f83f426ec2f1

https://github.com/Expensify/App/assets/43996225/f59ca7ea-2238-4e9e-8fa2-80dc17039f2f

**Expensify/Expensify Issue URL:**
**Issue reported by:** @natnael-guchima
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1683789139042789

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017527d879aa4e3b3e</li>
        <li>Upwork Job ID: 1659642295153512448</li>
        <li>Last Price Increase: 2023-05-19</li>
    </ul>
</details>

**Comment by mallenexpensify** [comment]
Gonna test tomorrow, hopping offline now

**Comment by mallenexpensify** [comment]
Am able to reproduce, @bondydaa , guessing this can be internal but wanted to double check.

**Comment by Natnael-Guchima** [comment]
Also, right clicking on the reacted emoji displays two accounts in the list - the default contact method and secondary login even though the reaction count is one. Here is a [link](https://expensify.slack.com/archives/C049HHMV9SM/p1683789750593049?thread_ts=1683789139.042789&cid=C049HHMV9SM) to the slack report I mentioned. It might have the same root cause to the issue being fixed here, and it might be good idea to address the issue here. 


![Screenshot_20230519-063318.jpg](https://github.com/Expensify/App/assets/57330233/bf14322f-cb38-480d-8dfb-e382e9230c38)

**Comment by bondydaa** [comment]
the bug defined in the OP is most likely a backend thing and needs to be internal. 

anything with display flickering is going to be a front end issue but I don't think the OP bug is related anything flickering so that should be a new bug report.

**Comment by bondydaa** [comment]
moving this to weekly.

We do store the accountID in the DB for which users have reacted so I don't think this is a problem with how we are storing the reactions https://github.com/Expensify/Auth/blob/f98583b7ef924cf3c7955a09067eefc5e63361a8/auth/command/AddEmojiReaction.cpp#L153-L154

It might just be something in onyx or the API that ends up grabbing both the old and new login to display so I'll take a look there to start, if it's just something in onyx then this might actually be able to be an external issue but I'll confirm first before making that decision.

**Comment by mallenexpensify** [comment]
Looks like Melvin wants this a daily.  @bondydaa want to provide an update when you're back online?

**Comment by bondydaa** [comment]
Hmm okay tested this out an reproduced it.

<img width="1197" alt="image" src="https://github.com/Expensify/App/assets/4073354/6a3dd844-6cea-436d-9a56-22714bc764a4">

Here is what the onyx data looked like before the primary login change: 
<img width="1132" alt="Screenshot 2023-06-02 at 11 00 29 AM" src="https://github.com/Expensify/App/assets/4073354/010f4deb-1439-4485-9146-e33d318fde4d">
<img width="1128" alt="Screenshot 2023-06-02 at 11 00 43 AM" src="https://github.com/Expensify/App/assets/4073354/8a6be4a8-3a34-49bf-b88e-525d02e26aac">


And afterwords its the same (I made sure to click the circle "refresh" button as well):
<img width="1129" alt="Screenshot 2023-06-02 at 11 09 11 AM" src="https://github.com/Expensify/App/assets/4073354/98d736a6-fa0c-4984-890e-d3cb86c32c0d">

Ahh I think I see, looking at the personalDetails onyx key after the update on the client for the user who didn't change their data shows:
<img width="1326" alt="image" src="https://github.com/Expensify/App/assets/4073354/82f00542-51c3-4a70-b2f1-ef15d60aaa19">

Where the same key for the client that did the primary login change shows: 
<img width="1362" alt="image" src="https://github.com/Expensify/App/assets/4073354/f09c7c39-7a02-4ac7-b1e5-0c82f15c6331">

So I bet we're not properly pushing the data to all clients and probably just the client that made the request. Will see if I can find this.

**Comment by bondydaa** [comment]
Yep looks like here is where we push the change when setting changing the primary login to the client that made the request https://github.com/Expensify/Web-Expensify/blob/345f8e86e9a5dd03d85b9ea8dc327937deca7267/lib/UserAPI.php#L1185-L1193

and here is what we do in new dot when we change the login https://github.com/Expensify/App/blob/b9069f7bc27eeefa0fc0cb8bb7173b78a5397ddc/src/libs/actions/User.js#L771-L854

So yeah I think we might need to update the API to trigger to update the `personalDetails` for users when change their primary login though this could end up being very expensive or trigger like thousands of clients to all hit the API quickly 🤔 

like if this is an #announce room that has 100 users, 1 user had reacted to a post and then changes their login that would need to be pushed to 99 other clients and they would have to fetch data from the servers.

So I'm also wondering if there is a way we could make it so that a client only goes to update these details when they need to.

**Comment by bondydaa** [comment]
discussing this a bit here https://expensify.slack.com/archives/C01GTK53T8Q/p1685730996869849 sounds like we might already have the functionality built out to push updates out for personal details changes here https://github.com/Expensify/Web-Expensify/blob/43f0a12209d2ab3ecb01fa11821792744abf3852/lib/PersonalDetailsAPI.php#L185-L189

just waiting on confirmation from others if we need to be worried about this happening at a large scale (like pushing out a login change in an announce room with 100s or 1000s of users).

**Comment by bondydaa** [comment]
Okay after a couple of discussion looks like we're not going to "pre optimize" here and worry about if this will fail when trying to send out to 100s+ clients, we have other things in the works to improve those connections. 

So I'll get a PR up to make sure we utilize the Personal Details API to push out these changes to the clients, not sure if anything will be needed on the front end or not, after we test my PR we'll confirm.

**Comment by bondydaa** [comment]
got a wip PR up here https://github.com/Expensify/Web-Expensify/pull/37736

**Comment by bondydaa** [comment]
pushed some updates to my PR but still need to fix up the tests and get them working.

**Comment by bondydaa** [comment]
still working on my PR, there is a decent size conflict I have to fix but deploy blocker and other chores got in the way so far this week.

**Comment by mallenexpensify** [comment]
commenting so it's not overdue

**Comment by bondydaa** [comment]
web deploy and deploy blockers have taken all my time this week, I have a PR in progress but haven't been able to get it finished

**Comment by bondydaa** [comment]
🙏 ing to the chore gods that I don't get another deployer week next week so i can finish off my PR.

**Comment by bondydaa** [comment]
i'll get back to my PR today, believe I just need to clean up the tests and get them working

**Comment by bondydaa** [comment]
still working on my PR

**Comment by bondydaa** [comment]
okay got a new PR up and tested, we shouldn't need anything in new dot either 🎉 https://github.com/Expensify/Web-Expensify/pull/38008

**Comment by bondydaa** [comment]
PR is in review, pushing updates for it today

**Comment by bondydaa** [comment]
okay the PR has been deployed to production. it was tested on staging without any problems but lets just double check and confirm that production new dot and production web api works as well.

@kavimuru can you have someone retest for us?

**Comment by bondydaa** [comment]
@s77rt I'm going to unassign you since I don't think we're needing any new dot changes here. probably should have done that a few weeks ago, sorry!

**Comment by bondydaa** [comment]
@mallenexpensify do you want to handle getting payment done for the bug report? I think that's the only thing we're going to need to pay here. but please let me know if I've forgotten/missed anything.

**Comment by Natnael-Guchima** [comment]
@bondydaa It seems to be working well on production too. 

cc: @kavimuru 

https://github.com/Expensify/App/assets/57330233/af5d9876-182a-4bf7-be27-a7356ab53b72

**Comment by bondydaa** [comment]
woohoo thanks!

**Comment by mallenexpensify** [comment]
@Natnael-Guchima , can you please accept the job and reply here once you have? 
https://www.upwork.com/jobs/~01a15e899abca5b634

**Comment by Natnael-Guchima** [comment]
@mallenexpensify I have accepted the offer. Thanks.

**Comment by mallenexpensify** [comment]
@Natnael-Guchima paid! I think we're good here so closing
